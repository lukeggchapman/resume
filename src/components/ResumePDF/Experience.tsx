import React from 'react'
import { View, Text, StyleSheet } from '@react-pdf/renderer'

import { ExperienceData } from './ResumeData'
import htmlToPDF from '../../utils/htmlToPDF'

interface ExperienceProps {
  data: ExperienceData
}

const styles = StyleSheet.create({
  content: {
    fontFamily: 'PT Sans',
    fontSize: 10,
  },
})

const sortGroupsByEndDate = (group: ExperienceData['group']) => {
  return group.sort((a, b) =>
    a.edges[0].node.frontmatter.endDate > b.edges[0].node.frontmatter.endDate
      ? 1
      : 0
  )
}

const Experience = ({ data }: ExperienceProps) => {
  const groupSorted = sortGroupsByEndDate(data.group)

  return (
    <>
      {groupSorted.map(group => {
        const { company } = group.edges[0].node.frontmatter

        return (
          <View key={company}>
            <Text>{company}</Text>
            {group.edges.map(edge => {
              const experience = edge.node
              const {
                endDateFormatted,
                startDate,
                endDate,
              } = experience.frontmatter

              return (
                <View style={styles.content} key={`${company}_${endDate}`}>
                  <Text>{`${startDate} - ${endDateFormatted}`}</Text>
                  <View>{htmlToPDF(experience.htmlAst)}</View>
                </View>
              )
            })}
          </View>
        )
      })}
    </>
  )
}

export default Experience
