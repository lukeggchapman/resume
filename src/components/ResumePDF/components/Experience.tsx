import React from 'react'

import hastToPDF from 'utils/hastToPDF'
import { ExperienceData } from '../resumeQuery'
import ContentItem from './common/ContentItem'

export interface ExperienceProps {
  data: ExperienceData
}

const Experience = ({ data }: ExperienceProps) => {
  return (
    <>
      {data.edges.map(({ node }) => {
        const {
          company,
          logo: { publicURL: logoURL },
          startDate,
          endDate,
          role,
        } = node.frontmatter

        return (
          <ContentItem
            key={company}
            title={company}
            subTitle={role}
            logo={logoURL}
            startDate={startDate}
            endDate={endDate}
          >
            {hastToPDF(node.htmlAst)}
          </ContentItem>
        )
      })}
    </>
  )
}

export default Experience
