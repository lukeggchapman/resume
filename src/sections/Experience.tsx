import React from 'react'

import hastToPDF from 'utils/hastToPDF'
import mdToHast from 'utils/mdToHast'
import ContentItem from 'components/ContentItem'

import assignarData from 'content/experience/2019-assignar.md'
import assignarLogo from 'content/experience/assignar.jpg'
import psyfariData from 'content/experience/2018-psyfari.md'
import psyfariLogo from 'content/experience/psyfari.png'
import learnosityData from 'content/experience/2015-learnosity.md'
import learnosityLogo from 'content/experience/learnosity.png'

const experienceData = [
  {
    frontmatter: assignarData.data,
    logo: assignarLogo,
    htmlAst: mdToHast(assignarData.content),
  },
  {
    frontmatter: psyfariData.data,
    logo: psyfariLogo,
    htmlAst: mdToHast(psyfariData.content),
  },
  {
    frontmatter: learnosityData.data,
    logo: learnosityLogo,
    htmlAst: mdToHast(learnosityData.content),
  },
]

const Experience = () => {
  return (
    <>
      {experienceData.map(({ frontmatter, logo, htmlAst }) => {
        const { company, startDate, endDate, role } = frontmatter

        return (
          <ContentItem
            key={company}
            title={company}
            subTitle={role}
            logo={logo}
            startDate={startDate}
            endDate={endDate}
          >
            {hastToPDF(htmlAst)}
          </ContentItem>
        )
      })}
    </>
  )
}

export default Experience
