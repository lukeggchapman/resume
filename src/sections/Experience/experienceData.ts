import mdToHast from 'utils/mdToHast'

import assignarData from 'content/experience/2019-assignar.md'
import assignarLogo from 'content/experience/assignar.jpg'
import learnosityData from 'content/experience/2015-learnosity.md'
import learnosityLogo from 'content/experience/learnosity.png'
import componoData from 'content/experience/2020-compono.md'
import componoLogo from 'content/experience/compono.png'

const experienceData = [
  {
    frontmatter: componoData.data,
    logo: componoLogo,
    htmlAst: mdToHast(componoData.content),
  },
  {
    frontmatter: assignarData.data,
    logo: assignarLogo,
    htmlAst: mdToHast(assignarData.content),
  },
  {
    frontmatter: learnosityData.data,
    logo: learnosityLogo,
    htmlAst: mdToHast(learnosityData.content),
  },
]

export default experienceData
