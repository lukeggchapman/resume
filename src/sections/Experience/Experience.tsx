import React from 'react'

import hastToPDF from 'utils/hastToPDF'
import ContentItem from 'components/ContentItem'
import Section from 'components/Section'

import experienceData from './experienceData'

const Experience = () => {
  return (
    <Section title="Experience">
      {experienceData.map(({ frontmatter, logo, htmlAst }) => {
        const { company, startDate, endDate, role } = frontmatter

        return (
          <ContentItem
            key={company}
            title={company}
            subTitle={role}
            logo={logo}
            startDate={startDate}
            endDate={endDate}
          >
            {hastToPDF(htmlAst)}
          </ContentItem>
        )
      })}
    </Section>
  )
}

export default Experience
