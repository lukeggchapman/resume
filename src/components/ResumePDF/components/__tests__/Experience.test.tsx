import React from 'react'
import { shallow } from 'enzyme'

import Experience, { ExperienceProps } from '../Experience'

jest.mock('../common/ContentItem', () => 'ContentItem')
jest.mock('utils/hastToPDF', () => jest.fn(htmlAst => htmlAst))

const baseProps: ExperienceProps = {
  data: {
    edges: [
      {
        node: {
          htmlAst: 'htmlAstTest1' as any,
          frontmatter: {
            company: 'companyTest1',
            endDate: 'endDateTest1',
            endDateFormatted: 'endDateFormattedTest1',
            logo: {
              publicURL: 'logoPublcUrlTest1',
            },
            role: 'roleTest1',
            startDate: 'startDateTest1',
            startDateFormatted: 'startDateFormattedTest1',
          },
        },
      },
      {
        node: {
          htmlAst: 'htmlAstTest2' as any,
          frontmatter: {
            company: 'companyTest2',
            endDate: 'endDateTest2',
            endDateFormatted: 'endDateFormattedTest2',
            logo: {
              publicURL: 'logoPublcUrlTest2',
            },
            role: 'roleTest2',
            startDate: 'startDateTest2',
            startDateFormatted: 'startDateFormattedTest2',
          },
        },
      },
    ],
  },
}

describe('Experience', () => {
  const render = () => shallow(<Experience {...baseProps} />)

  it('renders correctly', () => {
    expect(render()).toMatchSnapshot()
  })
})
