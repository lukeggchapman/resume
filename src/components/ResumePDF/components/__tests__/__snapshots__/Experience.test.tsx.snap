// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Experience renders correctly 1`] = `
<Fragment>
  <ContentItem
    endDate="endDateTest1"
    key="companyTest1"
    logo="logoPublcUrlTest1"
    startDate="startDateTest1"
    subTitle="roleTest1"
    title="companyTest1"
  >
    htmlAstTest1
  </ContentItem>
  <ContentItem
    endDate="endDateTest2"
    key="companyTest2"
    logo="logoPublcUrlTest2"
    startDate="startDateTest2"
    subTitle="roleTest2"
    title="companyTest2"
  >
    htmlAstTest2
  </ContentItem>
</Fragment>
`;
